homeassistant/components/frontier_silicon/config_flow.py

```diff
@@ -0,0 +1,235 @@
+        errors = {}
+
+        if user_input:
+            device_url = (
+                f"http://{user_input[CONF_HOST]}:{user_input[CONF_PORT]}/device"
+            )
+            try:
+                self._webfsapi_url = await AFSAPI.get_webfsapi_endpoint(device_url)
+            except FSConnectionError:
+                errors["base"] = "cannot_connect"
+            except Exception as exception:  # pylint: disable=broad-except
+                _LOGGER.exception(exception)
+                errors["base"] = "unknown"
+            else:
+                return await self._async_step_device_config_if_needed()
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_ssdp(self, discovery_info: ssdp.SsdpServiceInfo) -> FlowResult:
+        """Process entity discovered via SSDP."""
+
+        device_url = discovery_info.ssdp_location
+
+        speaker_name = discovery_info.ssdp_headers.get(SSDP_ATTR_SPEAKER_NAME)
+        self.context["title_placeholders"] = {"name": speaker_name}
+
+        try:
+            self._webfsapi_url = await AFSAPI.get_webfsapi_endpoint(device_url)
+        except FSConnectionError:
+            return self.async_abort(reason="cannot_connect")
+        except Exception as exception:  # pylint: disable=broad-except
+            _LOGGER.exception(exception)
+            return self.async_abort(reason="unknown")
+
+        # For manually added devices the unique_id is the radio_id,
+        # for devices discovered through SSDP it is the UDN
+        self._unique_id = discovery_info.ssdp_udn
+        await self.async_set_unique_id(self._unique_id)
+        self._abort_if_unique_id_configured(
+            updates={CONF_WEBFSAPI_URL: self._webfsapi_url}, reload_on_update=True
+        )
+
+        return await self._async_step_device_config_if_needed(show_confirm=True)
+
+    async def _async_step_device_config_if_needed(
+        self, show_confirm=False
+    ) -> FlowResult:
+        """Most users will not have changed the default PIN on their radio.
+
+        We try to use this default PIN, and only if this fails ask for it via `async_step_device_config`
+        """
+
+        try:
+            # try to login with default pin
+            afsapi = AFSAPI(self._webfsapi_url, DEFAULT_PIN)
+
+            self._name = await afsapi.get_friendly_name()
+
+            self.context["title_placeholders"] = {"name": self._name}
+
+            # _unique_id will already be set when discovered through SSDP with the SSDP UDN,
+            # however, when adding a device manually, it will still be empty at this point.
+            # Now we have successfully logged in, we can check the radio_id of this device
+            if self._unique_id is None:
+                self._unique_id = await afsapi.get_radio_id()
+                await self.async_set_unique_id(self._unique_id)
+                self._abort_if_unique_id_configured()
```

Comment:
Let's keep SSDP out of the equation for now.
```suggestion
            self._unique_id = await afsapi.get_radio_id()
            await self.async_set_unique_id(self._unique_id)
            self._abort_if_unique_id_configured()
```

Reply:
Not resolved (and should probably be moved out `try` into `else`.